### PR TITLE
fix(types): fix `StateCreator` subtyping

### PR DESCRIPTION
**Pull request by devanshj**
I originally had the phantom type because I assumed it would be required. I'm still a little skeptic about this change but just want to expermiment and see the CI results hence opening the PR.

This PR's branch is based on #1372's branch. Note that this does not require the change in #1372, but has it as base because if we're going with #1371 this PR would probably be much less useful.

### CODE REVIEW

File: tests/types.test.tsx [hunk 1]
```diff
@@ -188,3 +194,42 @@ it('state is covariant', () => {
...
+  }
+
+  const foo: () => StateCreator<State, []> = () => (set, get) => ({
+    count: 0,
+    increment: () => {
+      set({ count: get().count + 1 })
+    },
+  })
+
+  create<State>()(persist(foo()))
```
Comment (devanshj) [inline tests/types.test.tsx:231]: So fixing the `StateCreator` subtyping no longer errors on this line as it does today... https://tsplay.dev/mq3pjw

**Comment by devanshj** [comment]
Now that we're going with #1371 this is not much useful, we can revisit later if we want, closing for now.

**Comment by dai-shi** [comment]
> if we're going with https://github.com/pmndrs/zustand/pull/1371 this PR would probably be much less useful.

I get it, but removing `PopArgument` is something looking nice.

**Comment by devanshj** [comment]
I was initially a little skeptic because I misunderstood some nuance, now I'm less skeptic about this change, so we can reopen this if you want. (Ofc not the `UnknownBecauseOfGenericMutators` part but just removing the phantom parameter).

Not only it removes the `PopArgument` stuff but also improves the `StateCreator` in terms of subtyping so it's actually a good change.

Reopening, let me know what you think.

**Review by dai-shi — APPROVED** [review]
LGTM

**Pull request merged**